game/game/gamescript.cpp

```diff
@@ -2126,6 +2143,17 @@ std::string GameScript::npc_getnearestwp(std::shared_ptr<zenkit::INpc> npcRef) {
   return "";
   }
 
+std::string GameScript::npc_getnextwp(std::shared_ptr<zenkit::INpc> npcRef) {
+  auto npc     = findNpc(npcRef);
+  auto nearest = npc ? world().findWayPoint(npc->position()) : nullptr;
+  auto filter  = [&](const WayPoint& p) { return nearest != &p; };
+  auto wp      = nearest ? world().findWayPoint(npc->position(), filter) : nullptr;
+
+  if(wp)
```

Comment:
This function is very similar to `WayMatrix::findNextPoint`. for sake of similarity, probably better to move this code to `WayMatrix`.

Another thing to note, in filter you probably want to discard points that are already taken by other npc's. `p.isLocked()` can achive that.

Reply:
> ... I did some "nature studies" during my lunch break.

If vanilla doesn't check occupancy - than we good, only need to move code into `WayMatrix`.

Reply:
This one is still unresolved - hopefully this evening :-)

Reply:
> This one is still unresolved - hopefully this evening :-)
> only need to move code into WayMatrix.

Maybe I need to correct my previous comment: can you move implementation of of point search outside of `GameScript`? `World::findNextFreePoint` is a good reference.

Reply:
Move the implementation to World now, not sure what you think about the function naming, but we'll see :-)